src/test/ProtocolFeesImplementation.sol

```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.20;
+
+import {ProtocolFees} from "../ProtocolFees.sol";
+import {IProtocolFeeController} from "../interfaces/IProtocolFeeController.sol";
+import {PoolKey} from "../types/PoolKey.sol";
+import {Currency} from "../types/Currency.sol";
+import {PoolId, PoolIdLibrary} from "../types/PoolId.sol";
+import {Pool} from "../libraries/Pool.sol";
+import {Slot0} from "../types/Slot0.sol";
+
+contract ProtocolFeesImplementation is ProtocolFees {
+    using PoolIdLibrary for PoolKey;
+
+    mapping(PoolId id => Pool.State) internal _pools;
+
+    constructor(uint256 _controllerGasLimit) ProtocolFees(_controllerGasLimit) {}
+
+    function setPrice(PoolKey memory key, uint160 sqrtPriceX96) public {
```

Comment:
oops i didnt publish this question last night

Reply:
it's because in order to successfuly set a protocol fee on a pool, the pool needs to be initialized. and for a pool to be initialized, it needs to have a price set. i can add a comment to make that more clear